Sources/StreamChat/StateLayer/CurrentUser.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents the currently logged in user.
+@available(iOS 13.0, *)
+public struct CurrentUser {
```

Comment:
We also have the model `CurrentChatUser` - this would be a bid confusing.

Reply:
Yeah, that is a slight problem with all of this. Need to think if I can come up with a new name.

Reply:
The best I could come up with is `ConnectedUser` since it is related to the `ChatClient.connectUser()`

Reply:
I think it is a good idea 👍